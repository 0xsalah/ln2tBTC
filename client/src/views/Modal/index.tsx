import React from 'react'

interface ModalProps {
    isOpen: boolean
    title: string
    buttonText: string
    onButtonClick: () => void
}

const Modal: React.FC<ModalProps> = ({ isOpen, title, buttonText, children, onButtonClick }) => {
    return (
        isOpen ?
            <div className="jquery-modal blocker current">
                < div id="modal_code" className="modal modal-code" style={{ display: 'inline-block' }
                }>
                    <div className="modal__content">
                        <div className="modal__title">{title}</div>
                        <div className="modal__text">
                            {children}
                        </div>
                        <div className="modal__button row justify-end">
                            <button onClick={onButtonClick} className="btn" >{buttonText}</button>
                        </div>
                    </div>
                    <a href="#close-modal" rel="modal:close" className="close-modal ">Close</a>
                </div >
            </div >
            : null
    )
}

export default Modal