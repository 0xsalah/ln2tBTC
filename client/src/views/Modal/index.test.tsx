import React from "react";
import { render } from "@testing-library/react";
import Modal from ".";

test("snapshot of component rendered open", () => {
  const component = render(
    <Modal isOpen={true} buttonText="Ok" title="Test" onButtonClick={() => null} >
      <span>Test description</span>
    </Modal>
  );
  expect(component.baseElement).toMatchSnapshot();
});

test("snapshot of component rendered closed", () => {
  const component = render(
    <Modal isOpen={false} buttonText="Ok" title="Test" onButtonClick={() => null} >
      <span>Test description</span>
    </Modal>
  );
  expect(component.baseElement).toMatchSnapshot();
});