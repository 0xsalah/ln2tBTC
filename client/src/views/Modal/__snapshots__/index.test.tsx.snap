// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of component rendered closed 1`] = `
<body>
  <div />
</body>
`;

exports[`snapshot of component rendered open 1`] = `
<body>
  <div>
    <div
      class="blocker current"
      role="presentation"
    >
      <div
        class="modal"
        style="display: inline-block;"
      >
        <div
          class="modal__content"
        >
          <div
            class="modal__title"
          >
            Test
          </div>
          <div
            class=""
          >
            <div
              class="modal__text"
            >
              <span>
                Test description
              </span>
            </div>
          </div>
          <div
            class="modal__button row justify-end"
          >
            <button
              class="btn"
            >
              Ok
            </button>
          </div>
        </div>
        <a
          class="close-modal "
          href="#close-modal"
          rel="modal:close"
        >
          Close
        </a>
      </div>
    </div>
  </div>
</body>
`;
